xapian-core/include/xapian/weight.h

```diff
@@ -526,15 +526,20 @@ class XAPIAN_VISIBILITY_DEFAULT BM25Weight : public Weight {
      *		   document length normalisation of wdf is.  0 means no
      *		   normalisation; 1 means full normalisation.  (default 0.5)
      *
+     *  @param delta  A parameter for pseudo tf value to control the scale
+     * 		   of the tf lower bound. Delta(δ) can be tuned for e.g from 0.0 
+     * 		   to 1.5 but BM25+ can still work effectively across collections 
+     *		   with a fixed δ = 1.0. (default 1.0)
+     * 
      *  @param min_normlen  A parameter specifying a minimum value for
      *		   normalised document length.  Normalised document length
      *		   values less than this will be clamped to this value, helping
      *		   to prevent very short documents getting large weights.
      *		   (default 0.5)
      */
-    BM25Weight(double k1, double k2, double k3, double b, double min_normlen)
+    BM25Weight(double k1, double k2, double k3, double b, double delta, double min_normlen)
```

Comment:
I'd just refactor the old constructor to have an extra parameter at the end (as you have done), and implement the backwards compatible one in terms of that. Note that this is [a C++11 feature](https://en.wikipedia.org/wiki/C%2B%2B11#Object_construction_improvement), which we now accept (which is why the current one isn't done like that, I guess):

```c++
    BM25Weight(double k1, double k2, double k3, double b, double min_normlen)
        : BM25Weight(k1, k2, k3, b, min_normlen, 1.0) {}
```